source/kraken/data_manager.h

```diff
@@ -76,6 +77,8 @@ class DataManager {
         }
     }
 
+    std::unique_ptr<std::mutex> mutex = std::make_unique<std::mutex>();
```

Comment:
why use a unique_ptr<mutex> instead of mutex directly ?

Reply:
because `get_data` is const, I cannot lock a mutex non-mutable in `get_data`...

Reply:
otherwise, I must declare mutex as a mutable member which is disturbing for me

Reply:
ho noooo, the mutable is a weird by pass :flying_saucer: .
Otherwise, You can erase the const ?